app/models/manageiq/providers/ovirt/infra_manager/provision/cloning.rb

```diff
@@ -10,11 +10,8 @@ def destination_image_locked?
   end
 
   def find_destination_in_vmdb(ems_ref)
-    if source.template?
-      ManageIQ::Providers::Ovirt::InfraManager::Vm.find_by(:name => dest_name, :ems_ref => ems_ref)
-    else
-      ManageIQ::Providers::Ovirt::InfraManager::Template.find_by(:name => dest_name, :ems_ref => ems_ref)
-    end
+    assoc = source.template? ? :vms : :miq_templates
+    ems.send(assoc).find_by(:name => dest_name, :ems_ref => ems_ref)
```

Comment:
Super-minor, but we can avoid the `.send`, if you think that's nicer - up to you:

```suggestion
    assoc = source.template? ? ems.vms : ems.miq_templates
    assoc.find_by(:name => dest_name, :ems_ref => ems_ref)
```

Reply:
Yeah that's nicer